// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.tpc";
option java_outer_classname = "TPCProto";
option objc_class_prefix = "TPC";

package tpc;

// The coordinator service definition.
service Coordinator {
  // Receives work in the coordinator and sends it to the nodes
  rpc SendWork (WorkRequest) returns (WorkResponse) {}
}

//The node service definition
service Node {
  // Receives work in the node
  rpc ReceiveWork (WorkRequest) returns (WorkResponse) {}
}

message Transaction {
    string access = 1;
    string pk = 2;
    string column = 3;
    string action = 4;
}

// The request message containing the user's name.
message WorkRequest {
  repeated Transaction work = 1;
  string address = 2;
}

// The response message containing the greetings
message WorkResponse {
  string success = 1;
}
